enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,539 @@
+This framework relies on 3 independently developed pieces.
+
+1. [The kmods-via-containers code/config](https://github.com/kmods-via-containers/kmods-via-containers)
+
+Delivers the stencil code and configuration files for building and delivering kmods via containers. It also delivers a service `kmods-via-containers@.service` that can be instantiated for each instance of the KVC framework.
+
+2. The kernel module code that needs to be compiled
+
+This repo represents the kernel module code that contains the source code for building the kernel module. This repo can be delivered by vendors and generally knows *nothing* about containers. Most importantly, if someone wanted to deliver this kernel module via the KVC framework, the owners of the code don't need to be consulted. The project provides an [example kmod repo](https://github.com/kmods-via-containers/simple-kmod).
+
+3. A KVC framework repo for the kernel module to be delivered
+
+This repo defines a container build configuration as well as a library, userspace tools, and config files that need to be created on the host system. This repo does not have to be developed by the owner of the kernel module that is wanted to be delivered.
+
+It must define a few functions in the bash library:
+
+* `build_kmods()`
+  * Performs the kernel module container build
+* `load_kmods()`
+  * Loads the kernel module(s)
+* `unload_kmods()`
+  * Unloads the kernel module(s)
+* `wrapper()`
+  * A wrapper function for userspace utilities
+
+Customers can hook in their procedures on how to build, load, unload etc their kernel modules. We are providing only an interface not the actual "complicated" implementation of those steps (facade pattern). Customers can then use any tool(s) (akmods, dkms, ..) they need to build their modules. 
+
+[This repo](https://github.com/kmods-via-containers/kvc-simple-kmod) houses an example using `simple-kmod`.
+
+## Motivation
+
+In OpenShift v3.x out-of-tree drivers could be easily installed on the nodes, since the node was a full RHEL node with a subscription and needed tools were installed easily with yum. 
+
+In OpenShift v4.x this changed with the introduction of RHCOS. There are currently two different documented ways to enable out-of-tree drivers. One way is using a SRO based operator and the other is using kmods-via-containers. 
+
+We want to come up with a unified solution that works for our customers across RHCOS and RHEL. The solution should also help customers that are currently on RHEL7 to consider moving to RHCOS which is fully managed and easier to support in OpenShift.
+ 
+#### Day-2 DriverContainer Management OpenShift
+ 
+More and more customers/partners want to enable hardware and or software on OpenShift that need kernel drivers, which are currently not (and may never be) upstream and out-of-tree. 
```

Comment:
"currently not... out-of-tree" doesn't sound right.  Maybe rephrase to "which are currently (and maybe forever) out of tree and not upstream" or some such?